rxandroidble/src/main/java/com/polidea/rxandroidble2/internal/connection/MtuWatcher.java

```diff
@@ -1,26 +1,33 @@
 package com.polidea.rxandroidble2.internal.connection;
 
 
+import com.polidea.rxandroidble2.exceptions.BleDisconnectedException;
+import com.polidea.rxandroidble2.internal.RxBleLog;
+
 import bleshadow.javax.inject.Inject;
 import bleshadow.javax.inject.Named;
-
 import io.reactivex.Observable;
-import io.reactivex.disposables.SerialDisposable;
 import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Predicate;
 
 @ConnectionScope
 class MtuWatcher implements ConnectionSubscriptionWatcher, MtuProvider, Consumer<Integer> {
 
     private Integer currentMtu;
     private final Observable<Integer> mtuObservable;
-    private final SerialDisposable serialSubscription = new SerialDisposable();
 
     @Inject
     MtuWatcher(
             final RxBleGattCallback rxBleGattCallback,
             @Named(ConnectionComponent.NamedInts.GATT_MTU_MINIMUM) final int initialValue
     ) {
-        this.mtuObservable = rxBleGattCallback.getOnMtuChanged().retry();
+        this.mtuObservable = rxBleGattCallback.getOnMtuChanged()
+                .retry(new Predicate<Throwable>() {
+                    @Override
+                    public boolean test(Throwable throwable) throws Exception {
+                        return !(throwable instanceof BleDisconnectedException);
```

Comment:
Actually it could only be the `BleGattException` with `BleGattOperationType==ON_MTU_CHANGED`